piccolo/table.py

```diff
@@ -315,7 +315,7 @@ def __init_subclass__(
                 if isinstance(column, Email):
                     email_columns.append(column)
 
-                if isinstance(column, Secret):
+                if isinstance(column, Secret) or column._meta.secret:
```

Comment:
I don't think we need to check `isinstance(column, Secret)` - just `column._meta.secret` alone should work, because `Secret` columns should have `column._meta.secret = True`.